apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet/Trace/CloudTrace.cs

```diff
@@ -119,16 +117,11 @@ public static void Initialize(string projectId, HttpApplication application, Tra
         private void BeginRequest(object sender, EventArgs e)
         {
             TraceHeaderContext headerContext = TraceHeaderContextUtils.CreateContext(HttpContext.Current.Request);
-            TraceOptions headerOptions = _headerFactory.CreateOptions(headerContext);
-
             // If the trace header says to trace or if the rate limiter allows tracing continue.
-            if (!headerOptions.ShouldTrace)
+            if (headerContext.ShouldTrace == false || (headerContext.ShouldTrace == null &&
+                !_rateFactory.CreateOptions().ShouldTrace))
```

Comment:
Note that this is equivalent to

```csharp
return headerContext.ShouldTrace ?? _rateFactory.CreateOptions().ShouldTrace;
```

I personally find that easier to read as a sort of "defaulting" mechanism, but others will have different views...